feat: add initial zstd support

JamieMagee opened the pull request:
This adds support for `zstd` compressed archives.

This change is dependent on https://github.com/isaacs/minizlib/pull/35

Closes #438

isaacs commented on src/parse.ts:
```diff
@@ -442,7 +454,7 @@ export class Parser extends EE implements Warner {
         this[BUFFER] = undefined
       }
-      if (chunk.length < gzipHeader.length) {
+      if (chunk.length < Math.max(gzipHeader.length, zstdHeader.length)) {
```
The header length to test shoudl be set once, not calculated every time. Put something like this at the top level of this module:

```ts
const ZIP_HEADER_LEN = Math.max(gzipHeader.length, zstdHeader.length)
```

Then here:

```ts
if (chunk.length < ZIP_HEADER_LEN) {
```

JamieMagee commented:
> Main has the minizlib that has zstd support, wanna rebase and see if this works?

Everything works locally for me on Node 24

<details>
<summary>Details</summary>

```sh
$ npm run test

> tar@7.4.3 pretest
> npm run prepare


> tar@7.4.3 prepare
> tshy


> tar@7.4.3 test
> tap

 PASS  test/large-numbers.js 36 OK 3.300s
 PASS  test/cwd-error.js 1 OK 3.382s
 PASS  test/mode-fix.js 11 OK 3.251s
 PASS  test/header.js 39 OK 3.505s
 PASS  test/load-all.js 1 OK 3.709s
 PASS  test/map.js 5 OK 3.715s
 PASS  test/normalize-windows-path.js 4 OK 3.682s
 PASS  test/options.js 24 OK 3.682s
 PASS  test/index.js 5 OK 3.940s
 PASS  test/pack.js 681 OK 3.810s
 PASS  test/path-reservations.js 39 OK 6.860s
 PASS  test/make-command.ts 38 OK 10s
 PASS  test/pax.js 9 OK 7.011s
 PASS  test/list.ts 26 OK 10s
 PASS  test/read-entry.js 18 OK 6.907s
 PASS  test/strip-absolute-path.js 18 OK 7.088s
 PASS  test/symlink-error.js 1 OK 7.084s
 PASS  test/types.js 6 OK 7.046s
 PASS  test/strip-trailing-slashes.js 3 OK 7.201s
 PASS  test/get-write-flag.js 6 OK 11s
 PASS  test/warn-method.js 8 OK 1.596s
 PASS  test/winchars.js 3 OK 1.494s
 PASS  test/update.js 83 OK 2.081s
 PASS  test/write-entry.js 296 OK 1.932s
 PASS  test/replace.ts 104 OK 9.251s
 PASS  test/unpack.js 592 OK 9.742s
 PASS  test/writable-assignment-check.ts 1 OK 3.345s
 PASS  test/extract.ts 70 OK 14s
 PASS  test/create.ts 60 OK 16s
 PASS  test/normalize-unicode.js 65 OK 17s
 PASS  test/parse.js 18780 OK 24s


  🌈 TEST COMPLETE 🌈


Asserts:  21033 pass  0 fail  21033 of 21033 complete
Suites:      31 pass  0 fail        31 of 31 complete

# { total: 21033, pass: 21033 }
# time=32239.257ms
```

</details>

> Posted one little nit, but otherwise this looks great! I'll review more closely and get it landed once it's rebased on main and loading minizlib the normal way.

I addressed your comment about pre-calculating `ZIP_HEADER_LEN`.

Would you like me to add a guard for versions of Node that don't have support for zstd?
